editor/editor_settings.cpp

```diff
@@ -1654,7 +1654,7 @@ Ref<Shortcut> ED_SHORTCUT(const String &p_path, const String &p_name, uint32_t p
 		ie.instance();
 
 		ie->set_unicode(p_keycode & KEY_CODE_MASK);
-		ie->set_keycode(p_keycode & KEY_CODE_MASK);
+		ie->set_keycode(Key(p_keycode & KEY_CODE_MASK));
```

Comment:
Shouldn't `p_keycode` be of type `Key` too?

Reply:
Similarly to create_reference, this accepts unions of keys so uint32_t is the easiest option, and this isn't exposed. Otherwise we would need a lot of casts.

Reply:
I remember you can overload the `|` operations so that the result would be the type of the enum instead of int.

Reply:
@neikeq It doesn't work in the case of `+` for some reason. Operator code:

```cpp
inline Key operator+(Key a, Key b) {
	return (Key)((int)a + (int)b);
}
```

Output while compiling:

```
modules/visual_script/visual_script_editor.cpp:4529:83: error: invalid conversion from 'int' to 'Key' [-fpermissive]
 4529 |  ED_SHORTCUT("visual_script_editor/edit_member", TTR("Edit Member"), KEY_MASK_CMD + KEY_E);
      |                                                                      ~~~~~~~~~~~~~^~~~~~~
      |                                                                                   |
      |                                                                                   int
```

Reply:
Shouldn't one of the parameters be `KeyModifierMask`? You're doing that with the other operator overloads.

Reply:
@neikeq You're right, good catch! I fixed it.